### DISCUSSION TITLE
Enabling touch on CYD when using custom firmware

### DISCUSSION THREAD

**Discussion by BrazenCodebane** [witnessmenow/ESP32-Cheap-Yellow-Display#177] [Q&A]
I would appreciate some guidance on this subject. psuedocode or just the rough logic to solve the issue ,even more so appreciated.

Secondary question: 
is there a VM or simulator to test code for the CYD on via vscode or another IDE?

**Comment by MYCAMEL222** [comment]
Arduino IDE.....use TFT_eTouch.h

#include <TFT_eTouch.h>

#define ETOUCH_MOSI 32
#define ETOUCH_MISO 39
#define ETOUCH_SCK 25
#define ETOUCH_CS 33
#define ETOUCH_IRQ 0xff  // not used in this sketch

SPIClass hSPI(HSPI);  //define SPI port to be used by TFT_eTouch

TFT_eSPI tft = TFT_eSPI();
TFT_eTouch<TFT_eSPI> touch(tft, ETOUCH_CS, ETOUCH_IRQ, hSPI);
